### ISSUE TITLE
aria-hidden-focus does not account for inputs being inside a disabled <fieldset>

### ISSUE THREAD

**Issue by thomasbickley-bjss** [dequelabs/axe-core#2961]
Product: axe Extension

Expectation: An input that is inside a `<fieldset>` that has been disabled should not be treated as focusable by the aria-hidden-focus rule

Actual: An input that is inside a `<fieldset>` that has been disabled is still treated as focusable by the aria-hidden-focus rule

Motivation: The aria-hidden-focus rule returns a false positive when aria-hidden is used with a disabled fieldset

Example:
The aria-hidden-focus rule flags the following as a serious issue, despite the input not actually being focusable. 
```
<fieldset disabled="disabled" aria-hidden="true">
	<input />
</fieldset>
```

<pre><code>axe-core version: 4.2.0
axe extension version: 4.9.3
</code></pre>

**Comment by WilcoFiers** [comment]
Thank you for reporting the issue. We'll be sure to fix it. I think the problem is in dom.focusDisabled, which do
https://github.com/dequelabs/axe-core/blob/develop/lib/commons/dom/focus-disabled.js#L10

To fix this we'll need to have i check walk up the tree to see if there's a disabled fieldset. Should be fairly straightforward. The one thing to be mindful of in fixing this is to check how this interacts with shadow DOM. I recall that `disable` only works on descendants in the same DOM tree. We should test it to make sure.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base,
observed that, 
1. When given content made unfocusable through disabled fieldset`<fieldset id="target" disabled aria-hidden="true"><input /></fieldset>` test will pass
2. Fail when content is in the legend of a disabled fieldset
` <fieldset id="target" disabled aria-hidden="true"><legend><input /></legend></fieldset>`
3.also fail when content is in an aria-hidden but not disabled fieldset
`<fieldset id="target" aria-hidden="true"><input /></fieldset>`
![image](https://user-images.githubusercontent.com/47409579/128302122-ec206098-b767-4d82-b64e-aa77d21ffc79.png)

**Comment by straker** [comment]
No docs

**Comment by thomasbickley-bjss** [comment]
@straker I don't understand what you mean. What docs are missing here?

**Comment by straker** [comment]
Sorry, that was a paperwork style comment. Every issue needs testing and then might need docs updated. Was just saying that this issue doesn't need to update or add any docs.

**Comment by thomasbickley-bjss** [comment]
Was closing it accidental then?

**Comment by straker** [comment]
Nope. Closing was intended. The issue has been resolved and should be available in the latest release.
